### PR TITLE
test: added start/stop/delete/prune multiple containers from the container list test cases

### PR DESCRIPTION
**Pull request by danivilla9**
### What does this PR do?
This PR extends the container-smoke.spec.ts test suite to include the test cases of starting, stopping, deleting a container from the containers list, and pruning multiple containers on Podman Desktop.

### What issues does this PR fix or reference?
[https://github.com/containers/podman-desktop/issues/7577](https://github.com/containers/podman-desktop/issues/7577)

### How to test this PR?
Run the smoke tests with yarn test:e2e:smoke

### CODE REVIEW

File: tests/playwright/src/specs/container-smoke.spec.ts [hunk 1]
```diff
@@ -154,7 +154,61 @@ describe('Verification of container creation workflow', async () => {
...
+    const containersDetails = await containers.openContainersDetails(containerToRun);
+    await playExpect(containersDetails.heading).toBeVisible();
+    await playExpect(containersDetails.heading).toContainText(containerToRun);
+    // test state of container in summary tab
+    await pdRunner.screenshot('containers-container-details-stop-container.png');
+    const containerState = await containersDetails.getState();
+    playExpect(containerState).toContain(ContainerState.Running.toLowerCase());
+
+    await navigationBar.openContainers();
+    const containerRow = await containers.getContainerRowByName(containerToRun);
```
Comment (cbr7) [inline tests/playwright/src/specs/container-smoke.spec.ts]: there is the method containerExists() in the containers-page POM for this,

File: tests/playwright/src/specs/container-smoke.spec.ts [hunk 2]
```diff
@@ -154,7 +154,61 @@ describe('Verification of container creation workflow', async () => {
...
+    const containersDetails = await containers.openContainersDetails(containerToRun);
+    await playExpect(containersDetails.heading).toBeVisible();
+    await playExpect(containersDetails.heading).toContainText(containerToRun);
+    // test state of container in summary tab
+    await pdRunner.screenshot('containers-container-details-start-container.png');
+    const containerState = await containersDetails.getState();
+    playExpect(containerState).toContain(ContainerState.Exited.toLowerCase());
+
+    await navigationBar.openContainers();
+    const containerRow = await containers.getContainerRowByName(containerToRun);
```
Comment (cbr7) [inline tests/playwright/src/specs/container-smoke.spec.ts]: there is the method containerExists() in the containers-page POM for this,

File: tests/playwright/src/specs/container-smoke.spec.ts [hunk 3]
```diff
@@ -154,7 +154,61 @@ describe('Verification of container creation workflow', async () => {
...
+    if (containerRow === undefined) {
+      throw Error(`Container: '${containerToRun}' does not exist`);
+    }
+    const containerRowStartButton = containerRow.getByRole('button', { name: 'Start Container' });
+    await playExpect(containerRowStartButton).toBeVisible();
+    await containerRowStartButton.click();
```
Comment (cbr7) [inline tests/playwright/src/specs/container-smoke.spec.ts]: Please create a method that encapsulates this functionality and expose it fro ContainerPage POM.

File: tests/playwright/src/specs/container-smoke.spec.ts [hunk 4]
```diff
@@ -154,7 +154,61 @@ describe('Verification of container creation workflow', async () => {
...
+    playExpect(containerState).toContain(ContainerState.Running.toLowerCase());
+
+    await navigationBar.openContainers();
+    const containerRow = await containers.getContainerRowByName(containerToRun);
+    if (containerRow === undefined) {
+      throw Error(`Container: '${containerToRun}' does not exist`);
+    }
+    const containerRowStopButton = containerRow.getByRole('button', { name: 'Stop Container' });
+    await playExpect(containerRowStopButton).toBeVisible();
+    await containerRowStopButton.click();
```
Comment (cbr7) [inline tests/playwright/src/specs/container-smoke.spec.ts]: Please create a method that encapsulates this functionality and expose it fro ContainerPage POM.

File: tests/playwright/src/specs/container-smoke.spec.ts [hunk 5]
```diff
@@ -154,7 +154,61 @@ describe('Verification of container creation workflow', async () => {
...
 
-  test('Deleting a container', async () => {
+  test(`Start a container from the Containers list`, async () => {
+    const navigationBar = new NavigationBar(page);
+    const containers = await navigationBar.openContainers();
+    const containersDetails = await containers.openContainersDetails(containerToRun);
+    await playExpect(containersDetails.heading).toBeVisible();
+    await playExpect(containersDetails.heading).toContainText(containerToRun);
+    // test state of container in summary tab
+    await pdRunner.screenshot('containers-container-details-start-container.png');
```
Comment (cbr7) [inline tests/playwright/src/specs/container-smoke.spec.ts]: playwright tracing takes screenshots for everything, there is no need to manually call screenshot here or in any other place.

**Review by cbr7 — CHANGES_REQUESTED** [review]
Some comments left.

**Review by odockal — COMMENTED** [review]
Seems alright. Just work in @cbr7's feedback.

File: tests/playwright/src/model/pages/containers-page.ts [hunk 6]
```diff
@@ -49,7 +49,21 @@ export class ContainersPage extends MainPage {
     return new ContainerDetailsPage(this.page, name);
   }
 
-  async stopContainer(container: string): Promise<ContainerDetailsPage> {
+  async startContainerFromContainersList(containerRow: Locator): Promise<ContainersPage> {
```
Comment (odockal) [inline tests/playwright/src/model/pages/containers-page.ts]: I think that the function could be called simply `startContainer` and similarly for `stopContainer`, the context form where we call the function will make a distinction from details page. The same function will be in details page (not part of this PR). It will be then clear that we have something to refactor.

Comment (odockal) [reply tests/playwright/src/model/pages/containers-page.ts]: Also, it would be better to pass string param as a container name. Not the locator.

File: tests/playwright/src/specs/container-smoke.spec.ts [hunk 7]
```diff
@@ -154,7 +154,55 @@ describe('Verification of container creation workflow', async () => {
     await playExpect(startButton).toBeVisible();
   });
 
-  test('Deleting a container', async () => {
+  test(`Start a container from the Containers list`, async () => {
```
Comment (odockal) [inline tests/playwright/src/specs/container-smoke.spec.ts]: list -> page

File: tests/playwright/src/specs/container-smoke.spec.ts [hunk 8]
```diff
@@ -154,7 +154,55 @@ describe('Verification of container creation workflow', async () => {
...
+    const containersDetails = await containers.openContainersDetails(containerToRun);
+    await playExpect(containersDetails.heading).toBeVisible();
+    await playExpect(containersDetails.heading).toContainText(containerToRun);
+    // test state of container in summary tab
+    await navigationBar.openContainers();
+    await playExpect.poll(async () => await containers.containerExists(containerToRun)).toBeTruthy();
+
+    const containerRow = await containers.getContainerRowByName(containerToRun);
+    if (containerRow === undefined) {
+      throw Error(`Container: '${containerToRun}' does not exist`);
```
Comment (odockal) [inline tests/playwright/src/specs/container-smoke.spec.ts]: We do not throw an error, we let the test to fail by using assertions. You should remove this part as you have already added `containerExists` asssertion...

File: tests/playwright/src/specs/container-smoke.spec.ts (hunk 1 above)
Comment (odockal) [reply tests/playwright/src/specs/container-smoke.spec.ts]: This part should be removed, right?

**Review by odockal — CHANGES_REQUESTED** [review]

File: tests/playwright/src/specs/container-smoke.spec.ts (hunk 1 above)
Comment (cbr7) [reply tests/playwright/src/specs/container-smoke.spec.ts]: yes

File: tests/playwright/src/model/pages/containers-page.ts (hunk 6 above)
Comment (cbr7) [reply tests/playwright/src/model/pages/containers-page.ts]: As @odockal says, please refactor the method to receive the container name, the locator identification should be done behind the scene.

**Review by odockal — APPROVED** [review]

File: tests/playwright/src/model/pages/containers-page.ts [hunk 9]
```diff
@@ -49,7 +49,30 @@ export class ContainersPage extends MainPage {
...
   }
 
-  async stopContainer(container: string): Promise<ContainerDetailsPage> {
+  async startContainer(containerName: string): Promise<ContainersPage> {
+    const containerRow = await this.getContainerRowByName(containerName);
+    if (containerRow === undefined) {
+      throw Error(`Container: '${containerName}' does not exist`);
+    }
+    const containerRowStartButton = containerRow.getByRole('button', { name: 'Start Container' });
+    await playExpect(containerRowStartButton).toBeVisible();
```
Comment (cbr7) [inline tests/playwright/src/model/pages/containers-page.ts:58]:
```suggestion
    await playExpect(containerRowStartButton).toBeEnabled();
```

File: tests/playwright/src/model/pages/containers-page.ts [hunk 10]
```diff
@@ -49,7 +49,30 @@ export class ContainersPage extends MainPage {
...
+    return this;
+  }
+
+  async stopContainer(containerName: string): Promise<ContainersPage> {
+    const containerRow = await this.getContainerRowByName(containerName);
+    if (containerRow === undefined) {
+      throw Error(`Container: '${containerName}' does not exist`);
+    }
+    const containerRowStopButton = containerRow.getByRole('button', { name: 'Stop Container' });
+    await playExpect(containerRowStopButton).toBeVisible();
```
Comment (cbr7) [inline tests/playwright/src/model/pages/containers-page.ts:69]:
```suggestion
    await playExpect(containerRowStopButton).toBeEnabled();
```

**Comment by cbr7** [comment]
I'm not sure what is wrong but I am not able to run the tests on windows for this branch/PR, it's throws an error https://github.com/containers/podman-desktop-e2e/actions/runs/9872145579

**Review by cbr7 — APPROVED** [review]
